### PR TITLE
Nicer sauce

### PR DESCRIPTION
**Pull request by triblondon**
Thousands of lines of output made our sauce task hard to follow.  It now looks like this:

image

The clock icon indicates that we have not received data from that browser for that amount of time

### CODE REVIEW

**Comment by triblondon** [comment]
Some issues remain to resolve:
- [x] Some browsers regularly timeout before we start receiving results, notably Android
- [ ] Event loop blocking notifications get in the way of the console rewriting, so I disabled in all except prod, which is not ideal
- [x] Yet to fully test what happens when there are failures to report
- [x] Progress meter output fails pretty spectacularly when you try and do a compatgen because the tests don't fit on the console window

**Comment by triblondon** [comment]
Android: turns out this browser typically takes ~45 seconds to start up.  My timeout was simply too short.

File: tasks/grunt/saucelabs.js [hunk 1]
```diff
@@ -10,8 +10,8 @@ const testResultsFile = path.join(testResultsPath, 'results.json');
 const SauceTunnel = require('sauce-tunnel');
 const mkdirp = denodeify(require('mkdirp'));
 const cli = require('cli-color');
-const pollTick = 500;
-const testBrowserTimeout = 40000;
+const pollTick = 1000;
+const testBrowserTimeout = 60000;
```
Comment (JakeChampion) [inline tasks/grunt/saucelabs.js:14]: 👍

File: Gruntfile.js [hunk 2]
```diff
@@ -24,18 +24,25 @@ module.exports = function(grunt) {
 			}
 		},
 		"saucelabs": {
+			debug: {
+				options: {
+					urls: { all: 'http://127.0.0.1:3000/test/director?mode=all' },
```
Comment (JakeChampion) [inline Gruntfile.js]: I think this should grab port from the process.env if it exists as that is what the service will use.

File: Gruntfile.js [hunk 3]
```diff
@@ -24,18 +24,25 @@ module.exports = function(grunt) {
...
+					browsers: browsers.quick,
+					concurrency: 3
+				}
+			},
 			compat: {
 				options: {
 					urls: {
 						all: 'http://127.0.0.1:3000/test/director?mode=all',
```
Comment (JakeChampion) [inline Gruntfile.js]: same here, not sure why we didn't do this before. Needs to use process.env.PORT if it is set.

**Comment by JakeChampion** [comment]
From CircleCI:
```
 • android/5.1 (targeted): Error: Timed out at 'waiting for results'
```

**Comment by triblondon** [comment]
Dealt with Jake's feedback and my earlier issues.

**Review by JakeChampion — APPROVED** [review]
